src/subsidy.cpp

```diff
@@ -107,6 +111,26 @@ std::pair<NewsReferenceType, NewsReferenceType> SetupSubsidyDecodeParam(const Su
 	}
 	SetDParam(5, s->dst);
 
+	/* If the subsidy is being offered or awarded, add the right string to communicate the duration. */
+	if (mode == SUBSIDY_NEWS_OFFERED) {
```

Comment:
Probably turn this into a switch

Reply:
Still want a switch now that it's been simplified? I'm happy to add one, but this if/else chain doesn't seem too bad to me.